docs/user/reference/metrics/event.md

```diff
@@ -108,157 +67,345 @@ metrics.views.login_opened.record(
 )
 ```
 
-There are test APIs available too, for example:
+</div>
 
-```Python
-# Was any event recorded?
-assert metrics.views.login_opened.test_has_value()
-# Get a List of the recorded events.
-snapshot = metrics.views.login_opened.test_get_value()
-# Check that two events were recorded.
-assert 2 == len(snapshot)
-first = snapshot[0]
-assert "login_opened" == first.name
-# Check that no errors were recorded
-assert 0 == metrics.views.login_opened.test_get_num_recorded_errors(
-    ErrorType.INVALID_OVERFLOW
-)
+<div data-lang="Rust" class="tab">
+
+Note that an `enum` has been generated for handling the `extra_keys`: it has the same name as the event metric, with `Keys` added.
+
+```Rust
+use metrics::views;
+
+let mut extra = HashMap::new();
+extra.insert(views::LoginOpenedKeys::SourceOfLogin, "toolbar".into());
+views::login_opened.record(extra);
 ```
 
 </div>
 
-<div data-lang="C#" class="tab">
+<div data-lang="JavaScript" class="tab">
 
-Note that an `enum` has been generated for handling the `extra_keys`: it has the same name as the event metric, with `Keys` added.
+```js
+import * as views from "./path/to/generated/files/views.js";
+
+views.loginOpened.record({ sourceOfLogin: "toolbar" });
+```
+
+</div>
+
+<div data-lang="Firefox Desktop" class="tab">
+
+**C++**
+
+```c++
+#include "mozilla/glean/GleanMetrics.h"
+
+using mozilla::glean::views::LoginOpenedKeys;
+nsTArray<Tuple<LoginOpenedKeys, nsCString>> extra;
+nsCString source = "toolbar"_ns;
+extra.AppendElement(MakeTuple(LoginOpenedKeys::SourceOfLogin, source));
 
-```C#
-using static Mozilla.YourApplication.GleanMetrics.Views;
+mozilla::glean::views::login_opened.Record(std::move(extra))
+```
+
+**JavaScript**
```

Comment:
```suggestion
**Javascript**
```

Reply:
[TC39](https://tc39.es/) wants to have a word with you.